api/handlers.go

```diff
@@ -289,13 +303,42 @@ func (a *API) Unmount(w http.ResponseWriter, r *http.Request) {
 
 	volName := volConfig.String()
 
+	ut := &config.UseMount{
+		Volume:   volName,
+		Reason:   lock.ReasonMount,
+		Hostname: a.Hostname,
+	}
+
+	if !volConfig.Unlocked {
+		a.RemoveStopChan(volName)
+
+		// XXX to doubly ensure we do not UNMOUNT something that is held elsewhere
+		// (presumably because it is mounted THERE instead), we refuse to unmount
+		// anything that doesn't acquire a lock. We also remove the TTL refresh
+		// before taking it so it is not cleared in the unlikely event the mount
+		// takes longer than the TTL. Re-establish the TTL on error only if it is in
+		// locked mode.
+		if err := a.Client.PublishUse(ut); err != nil {
```

Comment:
it isn't a loop, to be clear. it never repeats.

the purpose of hte lock is to acquire it so nothing else can.

On Thu, Aug 4, 2016 at 2:29 PM, Yuva Shankar <notifications@github.com>
wrote:

> In api/handlers.go
> <https://github.com/contiv/volplugin/pull/386#discussion_r73606249>:
>
> > +	ut := &config.UseMount{
> > +		Volume:   volName,
> > +		Reason:   lock.ReasonMount,
> > +		Hostname: a.Hostname,
> > +	}
> > +
> > +	if !volConfig.Unlocked {
> > +		a.RemoveStopChan(volName)
> > +
> > +		// XXX to doubly ensure we do not UNMOUNT something that is held elsewhere
> > +		// (presumably because it is mounted THERE instead), we refuse to unmount
> > +		// anything that doesn't acquire a lock. We also remove the TTL refresh
> > +		// before taking it so it is not cleared in the unlikely event the mount
> > +		// takes longer than the TTL. Re-establish the TTL on error only if it is in
> > +		// locked mode.
> > +		if err := a.Client.PublishUse(ut); err != nil {
>
> okay. Now, I understand that the whole purpose of this loop is to ensure
> that lock is still there.
>
> If so, Isn't enough just to check if publishUse fails on "ErrExists"
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/contiv/volplugin/pull/386/files/f9835078301e384f8b75ae6aa6f2bd8838132c7a#r73606249>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AABJ6__9SyOskwF3ZLhxUsjeKfh7mD7sks5qclmigaJpZM4JcVWO>
> .
>